File: apps/React-TS/src/utils/types/devices.ts
/**
 * TypeScript types for the devices API response.
 * Mirrors the FastAPI Pydantic schemas in apps/FastAPI/schemas/devices.py.
 */

// ── GET /api/v1/devices ───────────────────────────────────────────────────────

export type DeviceStatus = 'connected' | 'disconnected' | 'unlinked';

export interface Device {
  name: string;
  status: DeviceStatus;
  hardware_uuid?: string | null;
  created?: string | null;
}

export interface DevicesResponse {
  devices: Device[];
}

File: apps/FastAPI/schemas/devices.py
"""
Pydantic schemas for the devices feature.

Endpoint → schema mapping:
  GET /api/v1/devices → DevicesResponse
"""

from __future__ import annotations

from enum import Enum
from typing import List, Optional

from pydantic import BaseModel


class DeviceStatus(str, Enum):
    connected    = "connected"
    disconnected = "disconnected"
    unlinked     = "unlinked"


class Device(BaseModel):
    """A single Tenovi device paired with the patient."""
    name: str
    status: DeviceStatus
    hardware_uuid: Optional[str] = None
    created: Optional[str] = None


class DevicesResponse(BaseModel):
    """
    List of devices paired with the patient via Tenovi.

    Mirrors the data used by:
      POST /healthie/iframe_provider_tab/devices
    """
    devices: List[Device]
